crypto/Pos/Binary/Crypto.hs

```diff
@@ -56,17 +58,35 @@ instance (Typeable algo, Typeable a, HashAlgorithm algo) => Bi (AbstractHash alg
 -- SecretSharing
 ----------------------------------------------------------------------------
 
-#define BiPvss(T, PT) \
-  instance Bi T where {\
-    encode = encodeBinary ;\
-    decode = decodeBinary };\
-  deriving instance Bi PT ;\
+instance Bi Scrape.PublicKey where
+    encode = encodeBinary
+    decode = decodeBinary
+
+deriving instance Bi C.VssPublicKey
+
+instance Bi Scrape.KeyPair where
+    encode = encodeBinary
+    decode = decodeBinary
+
+deriving instance Bi C.VssKeyPair
+
+instance Bi Scrape.Secret where
+    encode = encodeBinary
+    decode = decodeBinary
+
+deriving instance Bi C.Secret
+
+instance Bi Scrape.DecryptedShare where
+    encode = encodeBinary
+    decode = decodeBinary
+
+deriving instance Bi C.DecShare
+
+instance Bi Scrape.EncryptedSi where
+    encode = encodeBinary
+    decode = decodeBinary
 
-BiPvss (Scrape.PublicKey, C.VssPublicKey)
-BiPvss (Scrape.KeyPair, C.VssKeyPair)
-BiPvss (Scrape.Secret, C.Secret)
-BiPvss (Scrape.DecryptedShare, C.DecShare)
-BiPvss (Scrape.EncryptedSi, C.EncShare)
+deriving instance Bi C.EncShare
```

Comment:
Why?

Reply:
Should this not be there? I didn't add this myself. Could be a merge resolution error.

Reply:
No, I mean, why did you replace those CPP-generated instances with manually written ones? The CPP version was shorter and less error-prone. This is just some boilerplate code.

Reply:
I remember making this change in Lisbon, but I can't remember why. It's important to you that I put it back?

Reply:
No, not important. I just don't understand why it has been done, but it doesn't make anything significantly worse.